src/dlgProfilePreferences.cpp

```diff
@@ -325,47 +433,88 @@ dlgProfilePreferences::dlgProfilePreferences(QWidget* pF, Host* pH) : QDialog(pF
 
         // FIXME: Check this each time that it is appropriate for THIS build version
         comboBox_mapFileSaveFormatVersion->clear();
-        // Add default version:
-        comboBox_mapFileSaveFormatVersion->addItem(tr("%1 {Default, recommended}").arg(pHost->mpMap->mDefaultVersion), QVariant(pHost->mpMap->mDefaultVersion));
-        comboBox_mapFileSaveFormatVersion->setEnabled(false);
-        label_mapFileSaveFormatVersion->setEnabled(false);
-        if (pHost->mpMap->mMaxVersion > pHost->mpMap->mDefaultVersion || pHost->mpMap->mMinVersion < pHost->mpMap->mDefaultVersion) {
-            for (short int i = pHost->mpMap->mMinVersion; i <= pHost->mpMap->mMaxVersion; ++i) {
-                if (i == pHost->mpMap->mDefaultVersion) {
-                    continue;
-                }
+        // Handle case when no map is present:
+        if (pHost->mpMap && pHost->mpMap->mpRoomDB && pHost->mpMap->mpRoomDB->size() > 0 ) {
+            label_mapFileSaveFormatVersion->setEnabled(true);
+            Q_ASSERT_X((pHost->mpMap->mMaxVersion >= pHost->mpMap->mDefaultVersion
```

Comment:
I think a `static_assert()` would work well here

Reply:
See previous comment about the same Qt Macro elsewhere